src/zope/wfmc/process.py

```diff
@@ -920,14 +934,7 @@ def getValidOutgoingTransitions(process, activity_definition,
             # do not consider 'otherwise' transitions just yet
             otherwises.append(transition)
             continue
-        try:
-            active = transition.condition(process)
-        except:
-            if strict:
-                raise
-            else:
-                active = False
-        if active:
+        if checker(transition, process, strict):
```

Comment:
I suggest to remove `process` from the checker interface, leave it just `checker(transition)`. Process is available to caller anyway. Have a checker that does not deal with exception handler at all in zope.wfmc; remove that `strict` parameter completely, it doesn't make sense anymore: client applications can have their own checkers that handle exceptions if they need.